datahub-actions/src/datahub_actions/pipeline/pipeline.py

```diff
@@ -254,19 +263,20 @@ def _execute_transformer(
                 f"Caught exception while executing Transformer with name {type(transformer).__name__}"
             ) from e
 
-    def _execute_action(self, enveloped_event: EventEnvelope) -> None:
+    def _execute_action(self, enveloped_event: EventEnvelope) -> Optional[bool]:
         try:
-            self.action.act(enveloped_event)
+            retval = self.action.act(enveloped_event)
```

Comment:
So the "act" method already returns a boolean?

Reply:
Yeah, it accepts None or bool. Executor returns bool already, but if it's a legacy application not aware of this change and returning None, None will be converted to True, which means always commit (old behavior).